test/ColumnCollapserTest.py

```diff
@@ -0,0 +1,98 @@
+        self.assertEqual(float(m2['i_t_Foxog']), float(".505"))
+        self.assertEqual(float(m2['i_tumor_f']), float(".105"))
+        self.assertEqual(float(m2['hamilcar']), float("0.005"))
+        self.assertEqual(float(m2['barca']), float("0.01"))
+        self.assertEqual(m2['donotcollapse'], "100|4500")
+
+    def test_cannot_collapse(self):
+        """Make sure that we move on when we cannot collapse values."""
+        # m1 = MutationDataFactory.default_create(chr="1", start="10000", end="10000")
+        m1 = MutationData()
+        m1.createAnnotation('ALT_F2R1', "|36")
+        m1.createAnnotation('i_t_Foxog', "|")
+        m1.createAnnotation('i_tumor_f', "")
+        m1.createAnnotation('hamilcar', "0|blah")
+        m1.createAnnotation('barca', "carthage_rules")
+        m1.createAnnotation('donotcollapse', "1|45")
+
+        cc = ColumnCollapser()
+        cc.update_mutation(m1)
+        self.assertEqual(m1['ALT_F2R1'], "36")
+        self.assertEqual(m1['i_t_Foxog'], "")
+        self.assertEqual(m1['i_tumor_f'], "")
+        self.assertEqual(m1['hamilcar'], "0|blah")
+        self.assertEqual(m1['barca'], "carthage_rules")
+        self.assertEqual(m1['donotcollapse'], "1|45")
+
+    def test_updating_annotation_source(self):
+        """Test that a String can be passed in to update the annotation source if columns are collapsed"""
+        # m1 = MutationDataFactory.default_create(chr="1", start="10000", end="10000")
+        m1 = MutationData()
+        m1.createAnnotation('ALT_F2R1', "|36", annotationSource="TEST")
+        cc = ColumnCollapser()
+        cc.update_mutation(m1, "foo")
+        self.assertEqual(m1.getAnnotation("ALT_F2R1").getDatasource(), "foo")
+
+    def test_not_updating_annotation_source(self):
+        """Test that do not have to update annotation source if columns are collapsed"""
+        # m1 = MutationDataFactory.default_create(chr="1", start="10000", end="10000")
+        m1 = MutationData()
+        m1.createAnnotation('ALT_F2R1', "|36", annotationSource="TEST")
+        cc = ColumnCollapser()
+        cc.update_mutation(m1)
+        self.assertEqual(m1.getAnnotation("ALT_F2R1").getDatasource(), "TEST")
+
+    def test_annotation_copy(self):
+        """Test that we can create a backup annotation with the old values after collapsing, if requested."""
+        # m1 = MutationDataFactory.default_create(chr="1", start="10000", end="10000")
+        m1 = MutationData()
+        m1.createAnnotation('ALT_F2R1', "|36", annotationSource="TEST")
+        cc = ColumnCollapser()
+        cc.update_mutation(m1, new_annotation_source="foo", copy_old_suffix="_full")
+        self.assertEqual(m1["ALT_F2R1_full"], "|36")
+        self.assertEqual(m1["ALT_F2R1"], "36")
+        self.assertEqual(m1.getAnnotation("ALT_F2R1_full").getDatasource(), "TEST")
+        self.assertTrue(m1.getAnnotation("ALT_F2R1").getDatasource() != m1.getAnnotation("ALT_F2R1_full").getDatasource())
+
```

Comment:
Could you add a test demonstrating the behavior when the copied annotation conflicts with an existing annotation.  i.e. `depth` is collapsed but `depth_full` already exists.